std/stdio.d

```diff
@@ -513,8 +527,14 @@ Throws: `ErrnoException` in case of error.
             detach();
         }
 
-        _p = cast(Impl*) enforce(malloc(Impl.sizeof), "Out of memory");
+        _p = cast(Impl*) enforce(calloc(1, Impl.sizeof), "Out of memory");
```

Comment:
Any reason this problem couldn't be fixed simply by moving this line to after the `_fopen` call? As far as I can see, then the other changes will become unnecessary.

Reply:
This looks simpler solution.